pkg/controller/elasticsearch/driver/esstate.go

```diff
@@ -62,9 +62,7 @@ type memoizingNodes struct {
 
 // initialize requests Elasticsearch for nodes information, only once.
 func (n *memoizingNodes) initialize() error {
-	ctx, cancel := context.WithTimeout(n.ctx, esclient.DefaultReqTimeout)
-	defer cancel()
-	nodes, err := n.esClient.GetNodes(ctx)
+	nodes, err := n.esClient.GetNodes(context.Background())
```

Comment:
Why not reusing `n.ctx` here instead of creating a new context? Could be useful if later on we want to limit the total duration of a whole chain of calls.